packages/react-components/react-utilities/src/compose/overrideSlotClasses.ts

```diff
@@ -0,0 +1,9 @@
+import { mergeClasses } from '@griffel/react';
+import { SLOT_USER_CLASS_NAME_SYMBOL } from '../compose/constants';
+import { UnknownSlotProps } from '../compose/types';
+import { isSlot } from './isSlot';
+
+export const overrideSlotClasses = (slot: UnknownSlotProps, ...classNames: (string | false | undefined)[]) => {
+  const userClassName = isSlot(slot) ? slot[SLOT_USER_CLASS_NAME_SYMBOL] : undefined;
+  slot.className = mergeClasses(slot.className, ...classNames, userClassName);
+};
```

Comment:
Need to find the right package for this function to live. It depends on `@griffel/react`, so it can't be in react-utilities. There doesn't seem to be a great option for a package that depends on griffel, but isn't for a specific component. Maybe react-provider since it has the custom style provider, but I don't love it... 

Alternatively, we could provide a different function that doesn't directly call `mergeClasses`. I don't think it's a great idea to expose a `getSlotUserClassName` function, since we might not want to support that long-term. Perhaps a function that returns the argument array to be passed to mergeClasses, but that seems pretty ugly.

Reply:
we could also merge it ourselves though we would lose the extra work mergeClasses does and would have to restrict the classNames type to only string[]. Or let the user that provides the classnames call mergeClasses since they would have to do it from the custom style hooks anyways.
```ts
export const overrideSlotClasses = (slot: UnknownSlotProps, className: string) => {
  const userClassName = isSlot(slot) ? slot[SLOT_USER_CLASS_NAME_SYMBOL] : undefined;
  slot.className = `${slot.className} ${className} ${userClassName}`;
};

overrideSlotClasses(state, mergeClasses(class1, class2, class3))
```
this is not ideal but at least lets you move it to react-utilities.

Reply:
Is it that bad to add a dependency on griffel to react-utilities?  react-utilities-compat depends on griffel.

Reply:
Based on a discussion in tech sync, we're going to just export a getter for the original className.